### DISCUSSION TITLE
iOS使用PAGFile的replaceImage替换图片不成功是什么原因，播放后还是旧图

### DISCUSSION THREAD

**Discussion by chenshengwen** [Tencent/libpag#1965] [General] [answered]
<img width="993" alt="image" src="https://github.com/Tencent/libpag/assets/13128109/4808389c-b1b0-4b4d-a2d1-2714480aa72f">

**Comment by kevingpqi123** [comment, marked as answer]
1、 你这边使用的是 setPath 的方式，PAGImageView 会根据传入的 Path 自己创建一个 PAGFile，渲染的时候使用的是这个 PAGFile，而你替换图片的是自己创建的 PAGFile，和 PAGImageView 使用的 PAGFile 是两个完全不同的对象，所以肯定没作用哈，如果你需要替换占位图，不要使用 setPath 的方法，使用 setComposition，将你创建的 PAGFile 传入进去；
2、如果需要编辑文本和替换占位图，不推荐使用 PAGImageView，使用 PAGView，具体原因看下这篇文章：https://pag.art/docs/use-pagimageview.html

**Reply by chenshengwen** [reply]
感谢，已解决

**Reply by bqduan** [reply]
@kevingpqi123 你好，继楼上老哥的问题继续追问。现在是换成了PAGView，上下滑动页面列表，容易崩溃，请问有解决办法吗

**Comment by chenshengwen** [comment]
您的邮件我已收到，Thank  you！
